File: solutions/go/allergies/1/allergies.go
package allergies

import "slices"

var ALLERGENS = [8]string {
    "eggs", "peanuts", "shellfish", "strawberries",
    "tomatoes", "chocolate", "pollen", "cats",
}

func Allergies(allergies uint) []string {
	a := make([]string, 0, len(ALLERGENS))
    for i, v := range ALLERGENS {
        if allergies >> i & 1 == 1 {
            a = append(a, v)
        }
    }
    slices.Clip(a)
    return a
}

func AllergicTo(allergies uint, allergen string) bool {
	return allergies >> slices.Index(ALLERGENS[:], allergen) & 1 == 1
}

File: solutions/csharp/allergies/2/Allergies.cs
public enum Allergen
{
    Eggs,
    Peanuts,
    Shellfish,
    Strawberries,
    Tomatoes,
    Chocolate,
    Pollen,
    Cats
}

public record class Allergies(int mask)
{
    private static readonly Allergen[] ALLERGENS = Enum.GetValues<Allergen>();
    
    public bool IsAllergicTo(Allergen allergen) => (mask & 1 << (int) allergen) > 0;

    public Allergen[] List() => ALLERGENS.Where(allergen => IsAllergicTo(allergen)).ToArray();
}